tests/test_modeling_utils.py

```diff
@@ -1171,6 +1172,151 @@ def test_safetensors_torch_from_torch_sharded(self):
             self.assertTrue(torch.equal(p1, p2))
 
 
+@require_torch
+class ModelOnTheFlyConversionTester(unittest.TestCase):
+    @classmethod
+    def setUpClass(cls):
+        cls.user = "huggingface-hub-ci"
+        cls.repo_name = f"{cls.user}/test-model-on-the-fly"
+        cls._token = os.getenv("HUGGINGFACE_PRODUCTION_USER_TOKEN", None)
+
+        if cls._token is None:
+            raise ValueError("Cannot run tests as secret isn't setup.")
+
+        HfFolder.save_token(cls._token)
```

Comment:
In general, I would be careful about setting a token on the machine. If someone runs the test suite locally, their token will be replaced by this one which is annoying. I would suggest to either mock `HfFolder.get_token` for the whole class or to pass the token to `BertModel(...)` in the tests.

Also to [login](https://huggingface.co/docs/huggingface_hub/quick-start#login) a machine it's preferable to use `huggingface_hub.login(...)` which is more explicit (I try to slowly remove any use of `HfFolder`, except internally)